tests/config_test.go

```diff
@@ -514,4 +514,51 @@ var _ = Describe("[Serial][rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][le
 			})
 		})
 	})
+
+	Context("With a DownwardAPI defined", func() {
+
+		downwardAPIName := "downwardapi-" + uuid.NewRandom().String()
+		downwardAPIPath := config.GetDownwardAPISourcePath(downwardAPIName)
+
+		testLabelKey := "kubevirt.io.testdownwardapi"
+		testLabelVal := "downwardAPIValue"
+		expectedOutput := testLabelKey + "=" + "\"" + testLabelVal + "\""
+
+		It("[test_id:790]Should be the namespace and token the same for a pod and vmi", func() {
+			tests.SkipPVCTestIfRunnigOnKindInfra()
+
+			By("Running VMI")
+			vmi := tests.NewRandomVMIWithDownwardAPIAndLabel(downwardAPIName, map[string]string{testLabelKey: testLabelVal})
```

Comment:
It may make sense to keep in this case adding the downward api volume here instead of the helper. It kind of moves an important part of the test to the helper.

Reply:
@rmohr Maybe I missunderstood what you meant, but the volume is added in utils.go for all cases. The only difference is that for volume types that support multiple volumes (e.g., secret) the code also exposes an AddDisk public function so we can have a functest of adding multiple volumes for that type. 
This is the same as ServiceAcount volume for example.

Reply:
> @rmohr Maybe I missunderstood what you meant, but the volume is added in utils.go for all cases. The only difference is that for types that support multiple volumes (e.g., secret) the code also exposes an AddDisk public function so we can have a functest of adding multiple disks for that type.
> This is the same as ServiceAcount volume for example.

I think we should differentiate two things: 

 1) Tests which just need a volume of some type to test something else
 2) Tests which need a volume because the integration of that volume is under tests

For the second category it is from my perspective better to add the disk under question explicitly under the test to not split the test into two parts. That does not mean that you can't find examples in the code-base where we don't follow that rule.

I think that especially here we have a pretty clear example of a bad helper function. Either change the helper function name into something like `AddVMIWhichExposesLabelsViaDownwardsAPI` (still kind of strange that one does not just add the labels which one wants to see), or you set the important part on this test.

I hope that clarifies my line of thought a little.

Reply:
@rmohr Done. Using a local function in the test file to add a downwardAPI volume with label into a vmi